nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php

```diff
@@ -224,4 +252,28 @@ private function cwpLookup($postcode) {
     }
   }
 
+  /**
+   * Get postcode district from a Northern Ireland postcode.
+   *
+   * For Northern Ireland postcodes, the postcode district is always 1 or 2 digits following the postcode area 'BT'.
+   */
+  private function cwpGetPostcodeDistrict(string $postcode) {
+    $postcode_district = NULL;
+
+    // If postcode is a full NI postcode, or just the first part (outward code - e.g. BT1) ...
+    if (preg_match('/^[bB][tT][0-9]{1,2}( ?[0-9][a-zA-Z]{2})?$/', $postcode)) {
+      if (strlen($postcode) > 4) {
```

Comment:
I'd recommend you trim any spaces in `$postcode` here as a padding space between the two halves can affect your string length check and potentially some of the substring trim points could be offset by 1.

Reply:
I'd also consider lowercasing in the first instance so you can reduce the regex expression.

Reply:
@johangant , @omahm  - see latest commit.